src/dev/build/tasks/create_static_fs_with_node_modules_task.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import globby from 'globby';
+import normalizePath from 'normalize-path';
+import rimraf from 'rimraf';
+import { resolve } from 'path';
+import { generateStaticFsVolume } from 'static-fs';
+import { promisify } from 'util';
+
+const asyncRimRaf = promisify(rimraf);
+
+async function deletePathsList(list) {
+  for (const path of list) {
+    await asyncRimRaf(path, { glob: false });
```

Comment:
Are these required to run serially? If not, you can remove the `await` and they'll run concurrently. If so, maybe add a note about why so someone else doesn't cause an issue by "optimizing" this.